sources/functions/requestrr

```diff
@@ -0,0 +1,164 @@
+        s_base=$(sed -n 's|\(.*\)<UrlBase>\(.*\)</UrlBase>|\2|p' "$s_config")
+    fi
+    if [[ -f /install/.radarr.lock ]]; then
+        r_config="/home/$user/.config/Radarr/config.xml"
+        r_bind=$(sed -n 's|\(.*\)<BindAddress>\(.*\)</BindAddress>|\2|p' "$r_config")
+        r_port=$(sed -n 's|\(.*\)<Port>\(.*\)</Port>|\2|p' "$r_config")
+        r_api=$(sed -n 's|\(.*\)<ApiKey>\(.*\)</ApiKey>|\2|p' "$r_config")
+        r_base=$(sed -n 's|\(.*\)<UrlBase>\(.*\)</UrlBase>|\2|p' "$r_config")
+    fi
+    cat > /opt/requestrr/SettingsTemplate.json << CFG
+{
+  "Authentication": {
+    "Username": "$(_get_master_username)",
+    "Password": "${requestrr_pass}",
+    "PrivateKey": "${requestrr_key}"
+  },
+  "ChatClients": {
+    "Discord": {
+      "BotToken": "",
+      "ClientId": "",
+      "StatusMessage": "!help",
+      "TvShowRoles": [],
+      "MovieRoles": [],
+      "MonitoredChannels": [],
+      "EnableRequestsThroughDirectMessages": false,
+      "AutomaticallyNotifyRequesters": true,
+      "NotificationMode": "PrivateMessages",
+      "NotificationChannels": [],
+      "AutomaticallyPurgeCommandMessages": false,
+      "DisplayHelpCommandInDMs": true
+    }
+  },
+  "DownloadClients": {
+    "Ombi": {
+      "Hostname": "",
+      "Port": 3579,
+      "ApiKey": "",
+      "ApiUsername": "",
+      "BaseUrl": "",
+      "UseSSL": false,
+      "Version": "3"
+    },
+    "Overseerr": {
+      "Hostname": "",
+      "Port": 5055,
+      "ApiKey": "",
+      "DefaultApiUserID": "",
+      "UseSSL": false,
+      "Version": "1"
+    },
+    "Radarr": {
+      "Hostname": "${r_bind}",
+      "Port": ${r_port:-7878},
+      "ApiKey": "${r_api}",
+      "BaseUrl": "${r_base}",
+      "MovieProfileId": "1",
+      "MovieRootFolder": "",
+      "MovieMinimumAvailability": "",
+      "MovieTags": [],
+      "AnimeProfileId": "1",
+      "AnimeRootFolder": "",
+      "AnimeMinimumAvailability": "",
+      "AnimeTags": [],
+      "SearchNewRequests": true,
+      "MonitorNewRequests": true,
+      "UseSSL": false,
+      "Version": "2"
+    },
+    "Sonarr": {
+      "Hostname": "${s_bind}",
+      "Port": ${s_port:-8989},
+      "ApiKey": "${s_api}",
+      "BaseUrl": "${s_base}",
+      "TvProfileId": "1",
+      "TvRootFolder": "",
+      "TvTags": [],
+      "TvLanguageId": "1",
+      "TvUseSeasonFolders": true,
+      "AnimeProfileId": "1",
+      "AnimeRootFolder": "",
+      "AnimeTags": [],
+      "AnimeLanguageId": "1",
+      "AnimeUseSeasonFolders": true,
+      "SearchNewRequests": true,
+      "MonitorNewRequests": true,
+      "UseSSL": false,
+      "Version": "3"
+    }
+  },
+  "BotClient": {
+    "Client": "",
+    "CommandPrefix": "!"
+  },
+  "Movies": {
+    "Client": "Disabled",
+    "Command": "movie"
+  },
+  "TvShows": {
+    "Client": "Disabled",
+    "Command": "tv",
+    "Restrictions": "None"
+  },
+  "Port": 4545,
+  "BaseUrl" : "/requestrr",
+  "Version": "1.12.0"
```

Comment:
Holup, won't this change? This should probably not be here?

Reply:
Devs seem to have written an upgrade path for the config file in the past